### PR TITLE
New labs

### PR DESCRIPTION
**Pull request by palandreassen**

### CODE REVIEW

File: getting-started.rst [hunk 1]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+  - The "Output"-tab of the output pipe provides what's needed to retrieve the data.
+
+.. _getting-started-labs-25:
+
+Lab 25
+^^^^^^
+
+This lab covers:
+
+- **Export to SQL-database**
```
Comment (TarjeiSkrede) [inline getting-started.rst:2288]: Should be "Write data to a SQL-database".

File: getting-started.rst [hunk 2]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+
+SQL databases are common target systems for data export from Sesam. There are a few different SQL systems available in sesam, and all of them are used togother with the SQL sink when exporting data. For the following task you need an SQL database to write to. If you don't have one at hand, there are free online database services where you can set up a database using a trial account. https://www.elephantsql.com/ is one where you can set up a postgresql database.
```
Comment (TarjeiSkrede) [inline getting-started.rst:2290]: Typo: togother.

File: getting-started.rst [hunk 3]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+
+5. Send your data to the database table by starting the output pipe. Check the execution log of the pipe to see how it ran. These logs are the first place to look for clues of what went wrong in a failed run. Verify that the entities have been written to the table in your database. If you need to run a query to view the data in the database, you can run ``select * from person;``.
+
+Tips
+
+  - The system config will be different depending on what type of database you are connecting to. Selecting a template will help with config creation. The Developer Guide also have full descriptions of the possible parameters of your system config. In many cases copying the "Example config" from the docs or selecting a config template and filling in the values to match your database server and authentication will be sufficient.
+  - The SQL sink also has a lot of parameters for customizing your SQL operations, like the possibility to define a table's columns in ``schema_definition`` and creating the table on a first run or a reset of the output pipe. Only the parameters "type", "system" and "table" are required.
```
Comment (TarjeiSkrede) [inline getting-started.rst:2319]: Add info that this requires the system user to have the necessary permissions in the database.

File: getting-started.rst [hunk 4]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+  - The ``pump-mode`` environment variable is used to make sure an output pipe in a non-production environment does not export data unless the pipe is started manually. In a production environment ``pump-mode`` would be set as "scheduled" and the pipe's pump config would include a "cron-expression" or "schedule_interval" that specifies when to run. 
+
+.. _getting-started-labs-26:
+
+Lab 26
+^^^^^^
+
+This lab covers:
+
+- **Export to API**
```
Comment (TarjeiSkrede) [inline getting-started.rst:2329]: Same as above: "Write data to an API endpoint".

File: getting-started.rst [hunk 5]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+
+API's are another common external system type that Sesam integrates with. Let's try posting some data to jsonplaceholder.typicode.com, which is a API containing test data that we can also post data to. The data will not actually be saved and stored, but the status codes that the API returns are the same as if it did. The operation we want to imitate is described on this page https://jsonplaceholder.typicode.com/guide.html under "Create a resource".
+
+Task
+++++
+
+- POST entities to an API. Follow the steps below:
+
+1. Create a URL system in Sesam named "typicode" with the base url ``"https://sonplaceholder.typicode.com/%s"`` as its ``url_pattern``. Store the url as an environment variable and refer to that in the system config.
```
Comment (TarjeiSkrede) [inline getting-started.rst:2338]: Wrong url, should be: "https://jsonplaceholder.typicode.com/%s"

File: getting-started.rst [hunk 6]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+          ],
+          ["add", "municipality-ni",
+            ["ni", "geonorge-municipality", "_T.municipality"]
+          ]
+        ]
+      }
+    }
+  }
+
+Because the "municipality" attribute is a rename from a differently named attribute in the source we have used the "_T" variable to refer to it. Refering to "_S.KommuneNummer" would yield the same result.
```
Comment (TarjeiSkrede) [inline getting-started.rst:3251]: Add: "The embedded entities are removed in this code snippet for brevity. They are found in the lab task text.".

File: getting-started.rst [hunk 7]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+          ["add", "_id", "_S.fylkesnummer"],
+          ["add", "rdf:type",
+            ["ni", "geonorge", "county"]
+          ]
+        ]
+      }
+    }
+  }
+
+
```
Comment (TarjeiSkrede) [inline getting-started.rst:3295]: Add: "The embedded entities are removed in this code snippet for brevity. They are found in the lab task text.".

File: getting-started.rst [hunk 8]
```diff
@@ -1598,151 +1590,2166 @@ When communicating with the API we use requests methods such as **GET**, **POST*
...
+Results after running the pipe from the new pipe offset:
+
+.. image:: images/getting-started/labs/lab-change-tracking-lastseen-result.png
+    :width: 800px
+    :align: center
+    :alt: Only the three latest entities are updated
+
+Lab 23
+^^^^^^
+Preperation pipe config:
```
Comment (TarjeiSkrede) [inline getting-started.rst:3493]: Typo: preperation

**Review by grove — APPROVED** [review]

**Pull request merged**